### PR TITLE
Update/partial rewrite of editable-grid

### PR DESCRIPTION
**Pull request by silasporth**
While working on a project, I came across the editable-grid and wanted to use it. But I found some bugs that I couldn't fix unless I ported the whole structure to my project. Well, I did that because I liked the idea of the base so much and then I decided to fix the bugs here as well.

What I did:
- Added many Javadoc comments for a better understanding of the project
- Changed the visibility of most the methods for an increased extensibility
- Renamed a lot of files and methods to be more appropriate
- Updated the EditableDataTable to the newest version of Wicket Extensions' DataTable
- Fixed the sortability of EditableListDataProviders
- Redesigned the Add Feature with AddRowToolbar
- Refactored the editable columns
- Fixed the form validation done by EditableTableSubmitLink
- Ajax ready Toolbars
- Expanded the accessibilty of EditableGrid

If there are problems with any of these changes, I'm happy to help.

### CODE REVIEW

File: editable-grid-parent/editable-grid-examples/pom.xml [hunk 1]
```diff
@@ -20,6 +20,35 @@
 			<groupId>${project.groupId}</groupId>
 			<artifactId>wicketstuff-editable-grid</artifactId>
 		</dependency>
+
+		<!-- LOGGING DEPENDENCIES - SLF4J-SIMPLE -->
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-simple</artifactId>
+			<version>${slf4j.version}</version>
```
Comment (martin-g) [inline editable-grid-parent/editable-grid-examples/pom.xml:28]: Isn't that for testing too ?

**Review by martin-g — APPROVED** [review]
LGTM!

The API breaks may make someone unhappy though! An upgrade from 9.x to 10.x will be painful!

File: editable-grid-parent/editable-grid-examples/pom.xml [hunk 2]
```diff
@@ -20,6 +20,35 @@
...
+		</dependency>
+
+		<!--  JUNIT DEPENDENCY FOR TESTING -->
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter-engine</artifactId>
+			<version>5.9.3</version>
```
Comment (solomax) [inline editable-grid-parent/editable-grid-examples/pom.xml:35]: please remove versions from here (they are defined in `dependencyManagement` at parent pom

File: editable-grid-parent/editable-grid-examples/src/main/java/org/wicketstuff/egrid/page/HomePage.html [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-		<div id="ft">
-		</div>
-	</body>
+<head>
+    <meta charset="utf-8">
+    <title>Apache Wicket Quickstart</title>
+    <link href="style.css" rel="stylesheet" type="text/css">
+</head>
+<body>
+<div style="display: flex; justify-content: center;">
```
Comment (solomax) [inline editable-grid-parent/editable-grid-examples/src/main/java/org/wicketstuff/egrid/page/HomePage.html:9]: please avoid using of inline styles

**Review by solomax — CHANGES_REQUESTED** [review]
LGTM :)

with minor changes :)

File: editable-grid-parent/editable-grid-examples/pom.xml (hunk 1 above)
Comment (silasporth) [reply editable-grid-parent/editable-grid-examples/pom.xml:28]: Yes

**Review by solomax — APPROVED** [review]

**Comment by solomax** [comment]
Thanks for the contribution @silasporth !

**Comment by ilgrosso** [comment]
Hi @silasporth I am trying to understand how to update my code from 10.0.0-M1 to 10.0.0 (see #849) but having some issues: is there any usage sample around?

**Comment by silasporth** [comment]
> Hi @silasporth I am trying to understand how to update my code from 10.0.0-M1 to 10.0.0 (see #849) but having some issues: is there any usage sample around?

Hi, currently there is only the example in editable-grid-parent/editable-grid-examples but it should be working and it shows a minimal example.

**Comment by ilgrosso** [comment]
Thank you @silasporth I was able to get around it: https://github.com/apache/syncope/commit/f00ad780eab55ea9cfcf6caf9c5a822a73a76a41

**Pull request merged**